ui/raidboss/raidboss_config.ts

```diff
@@ -1416,7 +1416,7 @@ class RaidbossConfigurator {
   }
 }
 
-const flattenTimeline = (
+const pureFlattenTimeline = (
```

Comment:
I'm not sure I understand these `pureFlattenTimeline`/`flattenTimelinePlace` (in place?) changes.  The pure function isn't very pure as it's modifying `set`.  What's the goal here?

Reply:
Good catch, I missed it still modifying set